core/includes/menu.inc

```diff
@@ -2601,12 +2601,11 @@ function menu_link_get_preferred($path = NULL, $selected_menu = NULL) {
     $selected_menu = MENU_PREFERRED_LINK;
   }
 
-  if (!isset($preferred_links[$path])) {
+  if (!isset($preferred_links[$path]) && ($item = menu_get_item($path))) {
```

Comment:
`$item` may be NULL, attempting to access it as an array throws a warning, so we check that it is loaded before proceeding.